src/caterpillar_test.py

```diff
@@ -669,5 +678,127 @@ def test_meta_tags(self):
       self.assertIn('<meta content="test233" name="name"', js_file.read())
 
 
+class TestEndToEndConvert(TestCaseWithTempDir):
+  """Converts an entire Chrome App and checks the output is correct."""
+
+  def setUp(self):
+    """Converts a test Chrome App."""
+    super(TestEndToEndConvert, self).setUp()
+    self.input_dir = TTS_PATH
+    self.output_dir = os.path.join(self.temp_path, 'tts test output')
+    self.config_path = os.path.join(self.temp_path, 'config.json')
+
+    encoding = sys.getfilesystemencoding()
+
+    self.boilerplate_dir = 'caterpillar-📂'
+    self.report_dir = 'report ✓✓✓'
+    self.start_url = 'ttstest.html'
+    config_input = '{}\n{}\n{}\n'.format(
+        self.boilerplate_dir, self.report_dir, self.start_url).encode(encoding)
+
+    # Generate a config file using Caterpillar.
+    process = subprocess.Popen(
+        [CATERPILLAR_PATH, 'config', '-i', self.config_path],
+        stdin=subprocess.PIPE,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT)
+    process.communicate(input=config_input)
+    process.wait()
+
+    if not os.path.exists(self.config_path):
+      raise RuntimeError('Configuration file generation failed.')
+
+    process = subprocess.Popen(
+        [CATERPILLAR_PATH, 'convert', '-c', self.config_path, self.input_dir,
+         self.output_dir],
+        stdin=subprocess.PIPE,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT)
+    process.wait()
+
+  def test_no_unexpected_output_files(self):
```

Comment:
I think it makes sense to have one test that "the output matches the reference", as opposed to two separate tests. Just combine these two.